app/controllers/grade_entry_forms_controller.rb

```diff
@@ -204,7 +204,7 @@ def update_grade_entry_students
   # Download the grades for this grade entry form as a CSV file
   def download
     grade_entry_form = record
-    send_data grade_entry_form.export_as_csv,
+    send_data grade_entry_form.export_as_csv(@current_role),
```

Comment:
use `current_role` in controllers. You only need to access the instance variable directly in views because views don't have access to session methods.